Sources/Action.swift

```diff
@@ -18,12 +18,19 @@ import Result
 ///
 /// `Action` enforces serial execution, and disables the `Action` during the execution.
 public final class Action<Input, Output, Error: Swift.Error> {
-	private let deinitToken: Lifetime.Token
+	private enum Availability {
+		case enabledIdle
+		case enabledExecuting
+		case disabledIdle
+		case disabledExecuting
```

Comment:
The problem is that `isEnabled` and `isExecuting` are not independent. They together form a finite state machine, and IMO named cases are easier to understand than a switch with tuples of booleans.